cashapppay/src/main/java/com/adyen/checkout/cashapppay/internal/ui/model/CashAppPayComponentParamsMapper.kt

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (c) 2023 Adyen N.V.
+ *
+ * This file is open source and available under the MIT license. See the LICENSE file for more info.
+ *
+ * Created by oscars on 26/6/2023.
+ */
+
+package com.adyen.checkout.cashapppay.internal.ui.model
+
+import com.adyen.checkout.cashapppay.CashAppPayConfiguration
+import com.adyen.checkout.cashapppay.CashAppPayEnvironment
+import com.adyen.checkout.components.core.PaymentMethod
+import com.adyen.checkout.components.core.StoredPaymentMethod
+import com.adyen.checkout.components.core.internal.ui.model.ComponentParams
+import com.adyen.checkout.components.core.internal.ui.model.SessionParams
+import com.adyen.checkout.core.Environment
+import com.adyen.checkout.core.exception.ComponentException
+
+internal class CashAppPayComponentParamsMapper(
+    private val overrideComponentParams: ComponentParams?,
+    private val overrideSessionParams: SessionParams?,
+) {
+
+    fun mapToParams(
+        configuration: CashAppPayConfiguration,
+        sessionParams: SessionParams?,
+        paymentMethod: PaymentMethod,
+    ): CashAppPayComponentParams = configuration
+        .mapToParamsInternal(
+            clientId = paymentMethod.configuration?.clientId ?: throw ComponentException(
+                "Cannot launch Cash App Pay, clientId is missing in the payment method object."
+            ),
+            scopeId = paymentMethod.configuration?.scopeId ?: throw ComponentException(
+                "Cannot launch Cash App Pay, scopeId is missing in the payment method object."
+            ),
+            returnUrl = configuration.getReturnUrlOrThrow(sessionParams ?: overrideSessionParams),
+        )
+        .override(overrideComponentParams)
+        .override(sessionParams ?: overrideSessionParams)
+
+    fun mapToParams(
+        configuration: CashAppPayConfiguration,
+        sessionParams: SessionParams?,
+        @Suppress("UNUSED_PARAMETER") paymentMethod: StoredPaymentMethod,
+    ): CashAppPayComponentParams = configuration
+        // clientId and scopeId are not needed in the stored flow.
+        .mapToParamsInternal(null, null, configuration.getReturnUrlOrThrow(sessionParams ?: overrideSessionParams))
```

Comment:
Hmm this overriding logic is done outside of the `override` method right? Can we change that somehow?